opensquirrel/writer/writer.py

```diff
@@ -42,6 +42,7 @@ def visit_measure(self, measure: Measure) -> None:
     def visit_gate(self, gate: Gate) -> None:
         gate_name = gate.name
         if gate.is_anonymous:
+            gate_name = gate_name.replace("\n", "")
             self.output += f"{gate_name}\n"
```

Comment:
This code looks  quite weird. If you read it, you don't understand why all newlines are removed.

When implementing a string representation for a node, I'd recommend never to add a trailing newline in the callee, but in the caller. So, ultimately, if you want to print the whole IR, you would do something like `print(f"{}\n", ir). This way, you avoid the cases when you would like to get rid of the trailing newline.

Reply:
In the case of  a MatrixGate we want the newline escapes (\n) to be removed from the string because otherwise the array of the matrix does not print on a single line. I have added this explanation to the code.